# Workshop Schedule


## Introduction to R

### Day 1 

> Note: All times are based in U.S. Eastern Time

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 9:00 - 9:30 | [Workshop Introduction](../lectures/workshop_intro_slides.pdf) | Instructor |
| 9:30 - 10:45 | [Introduction to R and RStudio](https://hbctraining.github.io/Intro-to-R-flipped/lessons/01_introR-R-and-RStudio.html) | Instructor |
| 10:45 - 11:00 | BREAK | |
| 11:00 - 12:00 | [R Syntax and Data Structure](https://hbctraining.github.io/Intro-to-R-flipped/lessons/02_introR-syntax-and-data-structures.html) | Instructor |
| 12:00 - 1:00 | BREAK |
| 1:00 - 1:30 | [Functions and Arguments](https://hbctraining.github.io/Intro-to-R-flipped/lessons/03_introR-functions-and-arguments.html) | Instructor |
| 1:30 - 2:15 | [Reading in and Inpecting data](https://hbctraining.github.io/Intro-to-R-flipped/lessons/06_reading_and_data_inspection.html) | Instructor |
| 2:15 - 2:30 | BREAK | |
| 2:30 - 3:00 | [Function exercise](https://hbctraining.github.io/Intro-to-R-flipped/activities/Day2_activities.html) | Instructor |
| 3:00 - 4:00 | [Data Wrangling: Subsetting Vectors and Factors](https://hbctraining.github.io/Intro-to-R-flipped/lessons/05_introR-data-wrangling.html) | Instructor |

### Day 2

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 9:00 - 9:30 | [Packages and Libraries](https://hbctraining.github.io/Intro-to-R-flipped/lessons/04_introR_packages.html) | Instructor |
| 9:30 - 10:30 | [Data wrangling: data frames, matrices and lists](https://hbctraining.github.io/Intro-to-R-flipped/lessons/07_introR-data-wrangling2.html) | Instructor |
| 10:30 - 10:45 | BREAK | |
| 10:45 - 11:15 | [The %in% operator](https://hbctraining.github.io/Intro-to-R-flipped/lessons/08_identifying-matching-elements.html) | Instructor |
| 11:15 - 12:00 | [Reordering and matching](https://hbctraining.github.io/Intro-to-R-flipped/lessons/09_reordering-to-match-datasets.html) | Instructor |
| 12:00 - 1:00 | BREAK |
| 1:00 - 1:30 | [Setting up a data frame to plot (+ the `map()` function)](https://hbctraining.github.io/Intro-to-R-flipped/lessons/10_setting_up_to_plot.html) | Instructor |
| 1:30 - 2:15 | [Data wrangling exercises](https://hbctraining.github.io/Intro-to-R-flipped/activities/Day3_activities.html) | Instructor |
| 2:15 - 2:30 | BREAK | |
| 2:30 - 3:15 | [Introduction to ggplot2](https://hbctraining.github.io/Intro-to-R-flipped/lessons/11_ggplot2.html) | Instructor |
| 3:15 - 3:30 | [Using custom functions for consistent plots](https://hbctraining.github.io/Intro-to-R-flipped/lessons/011b_Custom_Functions_ggplot2.html) | Instructor |
| 3:30 - 4:00 | [Generating a boxplot with ggplot2](https://hbctraining.github.io/Intro-to-R-flipped/lessons/12_boxplot_exercise.html) | Instructor |

Gottas find a way to squeeze R into 2 days. I think  we will need to strip out lessons that aren't used in the scRNA-seq workshops. i.e. maybe the `match()` or purr functions? Maybe increase workshops to 5pm?

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 9:00 - 9:15 | [Writing to file and exporting plots](https://hbctraining.github.io/Intro-to-R-flipped/lessons/04_introR_packages.html) | Instructor |
| 9:15 - 9:45 | [Finding help](https://hbctraining.github.io/Intro-to-R-flipped/lessons/07_introR-data-wrangling2.html) | Instructor |
| 9:45 - 10:45 | [Tidyverse](https://hbctraining.github.io/Intro-to-R-flipped/lessons/08_identifying-matching-elements.html) | Instructor |
| 10:45 - 11:00 | BREAK | |
| 11:00 - 12:00 | [Tidyverse Exercises](https://hbctraining.github.io/Intro-to-R-flipped/activities/Day4_activities.html) | Instructor |


### Day 3

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 9:00 - 10:00 | [Introduction to Single Cell RNA-sequencing: a practical guide](https://github.com/hbctraining/Intro-to-scRNAseq/blob/master/slides/072624_Chan_scRNAseq_workshop_Arpita_Kulkarni.pdf) | Instructor |
| 10:00 - 10:30 | [scRNA-seq Background](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/01_intro_to_scRNA-seq.html) | Instructor |
| 10:45 - 11:00 | BREAK | |
| 10:30 - 11:15 | [Raw data to count matrix](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/02_SC_generation_of_count_matrix.html) | Instructor |
| 11:15 - 12:00 | [Quality Control Set-up](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/03_SC_quality_control-setup.html) | Instructor |
| 12:00 - 1:00 | BREAK |
| 1:00 - 1:15 | [Quality control of cellranger counts](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/04_cellranger_QC.html) | Instructor |
| 1:15 - 2:30 | [Quality control with additional metrics](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/04_SC_quality_control.html) | Instructor |
| 2:30 - 2:45 | BREAK | |
| 2:45 - 3:15 | [Theory of PCA](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/05_theory_of_PCA.html) | Instructor |
| 3:15 - 4:00 | [Normalization and regressing out unwanted variation](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/06_SC_SCT_normalization.html) | Instructor |

Might need longer for the "Normalization and regressing out unwanted variation" lesson

A summary of counts to matrix with alevin-fry like [Jeremy's talk](https://jeremymsimon.github.io/DS_scRNA_20240228/DS_scRNA_20240228.html) could be useful?

### Day 4

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 9:00 - 9:30 | [A brief introduction to Integration](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/06a_integration_cca_theory.html)) | Instructor |
| 9:30 - 10:15 | [Running CCA integration and complex integration tasks](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/06b_integration_code_harmony.html) | Instructor |
| 10:45 - 11:00 | BREAK | |
| 10:30 - 11:15 | [Clustering](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/02_SC_generation_of_count_matrix.html) | Instructor |
| 11:15 - 12:00 | [Clustering quality control Part 1](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/08_SC_clustering_quality_control.html) | Instructor |
| 12:00 - 1:00 | BREAK  |
| 1:00 - 1:30 | [Clustering quality control Part 2](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/08_SC_clustering_quality_control.html) | Instructor |
| 1:30 - 1:45 | [Seurat Cheatsheet](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/seurat_cheatsheet.html) | Instructor |
| 1:45 - 2:45 | [Marker Identification](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/09_merged_SC_marker_identification.html) | Instructor |
| 2:45 - 3:00 | BREAK | |
| 3:00 - 4:00 | [Pseudobulk set-up](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/01_setup_intro_dataset.html) | Instructor |

Workflow summary removed

### Day 5

| Time |  Topic  | Instructor |
|:-----------:|:----------:|:--------:|
| 9:00 - 10:00 | [Differential expression analysis using `FindMarkers()`](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/02_DEanalysis_using_FindMarkers.html)) | Instructor |
| 10:00 - 10:45 | [Aggregating counts by celltype using pseudobulk approach](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/03_pseudobulk_DESeq2.html) | Instructor |
| 10:45 - 11:00 | BREAK | |
| 11:00 - 12:00 | [DE analysis of pseudobulk data using DESeq2](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/04_pseudobulk_DE_analysis.html) | Instructor |
| 12:00 - 1:00 | BREAK  |
| 1:00 - 1:45 | [Visualization of differentially expressed genes](https://hbctraining.github.io/Intro-to-scRNAseq/lessons/05_pseudobulk_DE_visualizations.html) | Instructor |
| 1:45 - 3:00 | [Comparison of results from different DE approaches](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/06_DE_comparisons.html) | Instructor |
| 2:45 - 3:00 | BREAK | |
| 3:00 - 4:00 | [Functional Analysis](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/07_functional_analysis_pseudobulk.html) | Instructor |

Need time for:

[Methods for Differental Abundance](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/lessons/08_differential_abundance.html) 
Wrap-up



Would love to have [Alex's Spatial module](https://hbctraining.github.io/spatial_nanocourse/lessons/visium_hd.html) in this as well:


## Resources
TBD

***

*These materials have been developed by members of the teaching team at the [Harvard Chan Bioinformatics Core (HBC)](http://bioinformatics.sph.harvard.edu/). These are open access materials distributed under the terms of the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/) (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*

* *Some materials used in these lessons were derived from work that is Copyright Â© Data Carpentry (http://datacarpentry.org/). 
All Data Carpentry instructional material is made available under the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/) (CC BY 4.0).*
